src/common/vector/auxiliary_buffer.cpp

```diff
@@ -51,6 +51,9 @@ std::unique_ptr<AuxiliaryBuffer> AuxiliaryBufferFactory::getAuxiliaryBuffer(
         return std::make_unique<StringAuxiliaryBuffer>(memoryManager);
     case LogicalTypeID::STRUCT:
         return std::make_unique<StructAuxiliaryBuffer>(type, memoryManager);
+    case LogicalTypeID::RECURSIVE_REL:
```

Comment:
Should we switch physical dataType instead of physical dataType here?

Reply:
No we cannot, we need to know child type to initialize child vector